File: training/medium/there-is-no-spoon-episode-1.js
function recherche(sens, line, col){
    if(sens == "Horizontal"){
        for(var i = col+1; i<width; i++){
            if (arr[line][i] == "0"){
                return  " "+i+" "+line;
            }
        }
    } else {
        for(var j = line+1; j<height; j++){
            if (arr[j][col] == "0"){
                return  " "+col+" "+j;
            }
        }
    }
    return " -1 -1";
}

var width = parseInt(readline()); // the number of cells on the X axis
var height = parseInt(readline()); // the number of cells on the Y axis
var arr = [];

for (var i = 0; i < height; i++) {
    var line = readline(); // width characters, each either 0 or .
    arr.push(line);
}

for (var y=0; y<height;y++){
    for(var x=0; x<width;x++){
        if (arr[y][x] == "0"){
            res = " "+x+" "+y;
            res += recherche("Horizontal", y, x)
            res += recherche("Vertical", y, x)
            print(res) 
        }
    }
}
// Write an action using print()
// To debug: printErr('Debug messages...');
            

File: training/medium/there-is-no-spoon-episode-1.py
import sys
import math

width = int(input())  # the number of cells on the X axis
height = int(input())  # the number of cells on the Y axis

def recherche(sens, line, col):
    global arr , width, height
    if sens == "Horizontal":
        for x in range(col+1, width):
            if arr[line][x] == "0":
                return  " "+str(x)+" "+str(line)
    else:
        for y in range(line+1,height):
            if arr[y][col] == "0":
                return " "+str(col)+" "+str(y)
    return " -1 -1"

arr = []

for i in range(height):
    line = input()  # width characters, each either 0 or .
    arr.append(list(line))
            
for each_line in range(height):
    for each_col in range(width):
        if arr[each_line][each_col]=="0": #si c'est un noeud
            res = str(each_col)+" "+str(each_line)
            res += recherche("Horizontal", each_line, each_col)
            res += recherche("Vertical", each_line, each_col)
            print(res)  
            
# To debug: print("Debug messages...", file=sys.stderr)